lib/src/component2_suggestors/componentwillmount_migrator.dart

```diff
@@ -44,43 +46,104 @@ class ComponentWillMountMigrator extends GeneralizingAstVisitor
       return;
     }
 
-    if (extendsComponent2(containingClass)) {
-      // Update method name.
-      if (node.name.name == 'componentWillMount') {
+    var componentDidMountMethodDecl = containingClass.members
+        .whereType<MethodDeclaration>()
+        .firstWhere((member) => member.name.name == 'componentDidMount',
+            orElse: () => null);
+
+    if (extendsComponent2(containingClass) &&
+        node.name.name == 'componentWillMount') {
+      if (componentDidMountMethodDecl != null) {
+        if (node.body is BlockFunctionBody) {
+          var hasSuperCall = componentDidMountMethodDecl.body
+              .toSource()
+              .contains('super.componentDidMount();');
+          final block = (node.body as BlockFunctionBody).block;
+          var methodBody = sourceFile.getText(
+              block.leftBracket.end, block.rightBracket.offset);
+
+          // Add comment to check super calls if new super call will be added
+          // to `componentDidMount` method.
+          if (methodBody.contains('super.componentWillMount();\n') &&
+              !hasSuperCall &&
+              !hasComment(node, sourceFile, componentWillMountMessage)) {
+            yieldPatch(
+              componentDidMountMethodDecl.offset,
+              componentDidMountMethodDecl.offset,
+              '$componentWillMountMessage\n',
+            );
+          }
+
+          // Update or remove super call.
+          methodBody = methodBody.replaceAll(
+            RegExp(r'super\.componentWillMount\(\);\n?'),
+            hasSuperCall ? '' : 'super.componentDidMount();\n',
+          );
+
+          // Move body of `componentWillMount` to end of `componentDidMount`.
```

Comment:
@greglittlefield-wf @joebingham-wk @kealjones-wk @sydneyjodon-wk what is the reasoning behind moving the contents of `componentWillMount` to the **end** of `componentDidMount`?

In the React 15 / Component v1 lifecycle, the contents of `componentWillMount` would have executed before the contents of `componentDidMount`. Wouldn't moving stuff to the end reverse that order?

Reply:
That's a good point. I would think that you're right, and I can't think of why we specified it that way in the AC.

Reply:
Good catch; I believe that was a mistake we made while writing the AC, and you're correct in that it should be moved to the beginning of componentDidMount.